System/Random.hs

```diff
@@ -76,6 +76,8 @@ module System.Random
   , newStdGen
 
   -- * Random values of various types
+  , Uniform(..)
+  , UniformRange(..)
```

Comment:
Doesn't these classes essentially deprecate whole of `Random`, or what purpose `Random` would serve?

Reply:
Great question - here's a link to our thinking: https://github.com/idontgetoutmuch/random/issues/8

Also I have created a branch of our latest code with @Shimuuar's draft haddock here: https://github.com/idontgetoutmuch/random/tree/shimuuar-haddocks-atop-interface-to-performance - I find it easier to look at the haddocks.

Reply:
There's long discussion about this at #8

Reply:
@phadej `Random` vs `Uniform`+`UniformRange` describe different concepts. See Rust example of [`"Standard"`](https://rust-random.github.io/rand/rand/distributions/struct.Standard.html) distribution, which would correspond to what I am suggesting we can use `Random` class for.

Another huge reason to keep `Random` around is for backwards compatibility. There is no need to break thousands of packages out there in a wild, just because we added a slightly more theoretically sound of generating values.

Reply:
`Random` is not quite like rust's `Standard`. Existence of `randomR` break things somewhat. It breaks things for anything without sane notion of range. Take for example AST. Generating _some_ syntax tree is possible. But what generation of AST in range even mean?

Reply:
I agree about `randomR` that is why I suggest to deprecate it. Only `random` and `randomM` should stay undeprecated as part of `Random`.